analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -337,4 +346,95 @@ private void writeRepositorySettingsFile(Collection<ArtifactRepository> reposito
             getProject().getLogger().info("Repository export disabled.");
         }
     }
+
+    //TODO this needs to be improved since it is very naive
+    private void findDynamicDependencies(org.gradle.api.artifacts.Configuration configuration,
+            Set<ProjectRef> dynamicDependencies, Map<ProjectRef, String> dynamicDepToKey) {
+        configuration.getIncoming().getDependencies().withType(ExternalDependency.class).matching(d -> {
+            final String requiredVersion = d.getVersionConstraint().getRequiredVersion();
+            return requiredVersion.endsWith("latest.release") || requiredVersion.endsWith("+");
+        }).forEach(d -> {
+            final SimpleProjectRef ref = new SimpleProjectRef(d.getGroup(), d.getName());
+            dynamicDependencies.add(ref);
+            dynamicDepToKey.put(ref,
+                    new SimpleProjectVersionRef(ref, d.getVersionConstraint().getRequiredVersion()).toString());
+        });
+    }
+
+    private Set<ProjectVersionRef> updateDynamicDependenciesToExistingVersions(Set<ProjectVersionRef> allDependencies,
+            Set<ProjectRef> dynamicDependencies, Project project) {
```

Comment:
Is this to try and pass existing redhat dependency versions to DA for realignment? I am wondering if we should try and split this up (in the sense of multiple PRs) as this is covering too much

Reply:
This is done in order to retrieve the previously aligned version, otherwise repeat builds won't work.

I up to splitting it however you like .

Reply:
A repeat build should work due to the lock file - but a repeat alignment run is another question? I guess the question is whether rerunning alignment, should reuse previously locked dynamic versions or not ? (I don't think concrete versions are an issue) ?

Reply:
I would assume that a repeat build should use the same upstream version, but attempt to find a new aligned version for that dependency, correct?
That is what currently happens, and according to the tests, it works (the Undertow version is bumbed to redhat-00002)